module/shop_beta/shop_core.py

```diff
@@ -0,0 +1,137 @@
+from module.base.button import ButtonGrid
+from module.base.decorator import cached_property
+from module.logger import logger
+from module.ocr.ocr import Digit
+from module.shop.assets import *
+from module.shop.base import ShopBase, ShopItemGrid
+
+OCR_SHOP_CORE = Digit(SHOP_CORE, letter=(239, 239, 239), name='OCR_SHOP_CORE')
+OCR_SHOP_AMOUNT = Digit(SHOP_AMOUNT, letter=(239, 239, 239), name='OCR_SHOP_AMOUNT')
+
+
+class CoreShop(ShopBase):
+    _shop_core = 0
+
+    @cached_property
+    def shop_filter(self):
+        """
+        Returns:
+            str:
+        """
+        return self.config.CoreShop_Filter.strip()
+
+    @cached_property
+    def shop_items(self):
+        """
+        Returns:
+            ShopItemGrid:
+        """
+        shop_grid = self.shop_grid
+        shop_core_items = ShopItemGrid(shop_grid, templates={}, amount_area=(60, 74, 96, 95))
+        shop_core_items.load_template_folder('./assets/shop/core')
+        shop_core_items.load_cost_template_folder('./assets/shop/cost')
+        return shop_core_items
+
+    def shop_currency(self):
+        """
+        Ocr shop core currency
+
+        Returns
+            int: core amount
+        """
+        self._shop_core = OCR_SHOP_CORE.ocr(self.device.image)
+        logger.info(f'Core: {self._shop_core}')
+        return self._shop_core
+
+    def shop_check_item(self, item):
+        """
+        Args:
+            item: Item to check
+
+        Returns:
+            bool:
+        """
+        if item.price > self._shop_core:
+            return False
+        return True
+
+    def shop_buy_amount_execute(self, item):
+        """
+        Args:
+            item: Item to check
+
+        Returns:
+            bool: implicating failed to execute
+        """
+        index_offset = (40, 20)
+        limit = 0
+
+        # In case either -/+ shift position, use
+        # shipyard ocr trick to accurately parse
+        self.appear(AMOUNT_MINUS, offset=index_offset)
+        self.appear(AMOUNT_PLUS, offset=index_offset)
+        area = OCR_SHOP_AMOUNT.buttons[0]
+        OCR_SHOP_AMOUNT.buttons = [(AMOUNT_MINUS.button[2] + 3, area[1], AMOUNT_PLUS.button[0] - 3, area[3])]
+
+        # Total number that can be purchased
+        # altogether based on clicking max
+        # Needs small delay for stable image
+        self.appear_then_click(AMOUNT_MAX)
+        self.device.sleep((0.3, 0.5))
+        self.device.screenshot()
+        limit = OCR_SHOP_AMOUNT.ocr(self.device.image)
+        if not limit:
+            self.device.click(SHOP_CLICK_SAFE_AREA)  # Close amount prompt
+            return False
+
+        # Adjust purchase amount if needed
+        while 1:
+            if (limit * item.price) <= self._shop_core:
+                break
+            else:
+                limit -= 1
```

Comment:
Yes true can just use arithmetic to get this but I think needs to be like this since that would just get the total that can be acquired using the whole wallet and need to adjust the limit if needed.
```
total = int(self._shop_core // item.price)
diff = limit - total
if diff > 0:
    limit = total
```